Makefile

```diff
@@ -141,6 +150,16 @@ ifndef FLAKE8_PROG
 endif
 	@python -m flake8
 
+# build python wheel package for the available python version
+wheel:
+ifndef WHEEL_MODULE_PROG
+	$(error $(newline)$(ccred)python wheel package is not installed$(newline)$(ccend)\
+	$(ccblue)	https://packaging.python.org/guides/distributing-packages-using-setuptools/#wheels$(newline)$(ccend)\
+	$(ccred)You can install this package with$(newline)$(ccend)\
+	$(ccblue)	pip install wheel$(newline)$(ccend))
+endif
+	@python setup.py bdist_wheel
```

Comment:
If the command `pip wheel . --no-deps` is used instead of `python setup.py bdist_wheel`, `pip` will take care of installing all the necessary build dependencies into an isolated build environment in which the package and wheel will be built.

For example:
```
# Create virtualenv without setuptools and wheel installed into it
$ python -m virtualenv .venv --no-wheel --no-setuptools
created virtual environment CPython3.9.6.final.0-64 in 115ms
  creator CPython3Posix(dest=/home/jcmorin/jcmenv/aswf/OpenTimelineIO/.venv, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, via=copy, app_data_dir=/home/jcmorin/.local/share/virtualenv)
    added seed packages: pip==21.2.2
  activators BashActivator,CShellActivator,FishActivator,PowerShellActivator,PythonActivator

# Activate the virtualenv
$ source .venv/bin/activate

# Prove that dependencies are really not present
(.venv) $ which wheel
which: no wheel in (/home/jcmorin/jcmenv/aswf/OpenTimelineIO/.venv/bin:/opt/google-cloud-sdk/bin:/home/jcmorin/.krew/bin:/opt/dart-sass/1.32.2:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/jcmorin/.yarn/bin:/opt/hugo/0.84.1:/opt/android-studio/4.0.1/bin:/opt/flutter/1.22.3/bin:/opt/kubectl/v1.21.0:/opt/kustomize/4.0.5)
(.venv) $ python -c 'import wheel'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'wheel'
(.venv) $ python -c 'import setuptools'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ModuleNotFoundError: No module named 'setuptools'

# Now build the package (using --no-deps because I don't want to create wheels for our direct runtime dependencies (pyaaf2 for example).
(.venv) $ pip wheel . -v --no-deps
Processing /home/jcmorin/jcmenv/aswf/OpenTimelineIO
  DEPRECATION: A future pip version will change local packages to be built in-place without first copying to a temporary directory. We recommend you use --use-feature=in-tree-build to test your packages with this new behavior before it becomes the default.
   pip 21.3 will remove support for this functionality. You can find discussion regarding this at https://github.com/pypa/pip/issues/7555.
  Running command /home/jcmorin/jcmenv/aswf/OpenTimelineIO/.venv/bin/python /tmp/pip-standalone-pip-5fpo8xk4/__env_pip__.zip/pip install --ignore-installed --no-user --prefix /tmp/pip-build-env-pjbgmhwh/overlay --no-warn-script-location --no-binary :none: --only-binary :none: -i https://pypi.org/simple -- setuptools wheel 'cmake>=3.12'
  Collecting setuptools
    Using cached setuptools-57.4.0-py3-none-any.whl (819 kB)
  Collecting wheel
    Using cached wheel-0.36.2-py2.py3-none-any.whl (35 kB)
  Collecting cmake>=3.12
    Using cached cmake-3.21.1.post1-py2.py3-none-manylinux_2_5_x86_64.manylinux1_x86_64.whl (19.8 MB)
  Installing collected packages: wheel, setuptools, cmake
  Successfully installed cmake-3.21.1.post1 setuptools-57.4.0 wheel-0.36.2
  WARNING: You are using pip version 21.2.2; however, version 21.2.3 is available.
  You should consider upgrading via the '/home/jcmorin/jcmenv/aswf/OpenTimelineIO/.venv/bin/python -m pip install --upgrade pip' command.
  Installing build dependencies ... done
......
...
..
output continues
```

This is made possible thanks to the `pyproject.toml` that we have in the root of the repository. In it, we define the dependencies required to build the package (including cmake) and pip takes care of creating an environment in which the package can be built with the specified dependencies.

Reply:
Ohh that's nice! I was a bit confused about some dependencies "automagically" being available.
I'll have a look and simplifie this.